source/localizable/tutorial/service.md

```diff
@@ -0,0 +1,293 @@
+      element = this.createMapElement();
+      this.get('mapUtil').createMap(element, location);
+      this.set(`cachedMaps.${camelizedLocation}`, element);
+    }
+    return element;
+  },
+
+  createMapElement() {
+    let element = document.createElement('div');
+    element.className = 'map';
+    element.setAttribute('style', 'width:300px;height:300px;');
+    return element;
+  }
+
+});
+```
+
+### Making Google Maps Available
+
+Before implementing the map utility we need to make the 3rd party map API available to our Ember app.
+There are several ways to include 3rd party libraries in Ember. See the guides section on [managing dependencies](../../addons-and-dependencies/managing-dependencies/) as a starting point when you need to add one.
+
+Since Google provides its map API as a remote script, we'll use curl to download it into our project's vendor directory.
+
+From your project's root directory, run the following command to put the Google maps script in your projects vendor folder as `gmaps.js`.
+
+```shell
+curl -o vendor/gmaps.js https://maps.googleapis.com/maps/api/js?v=3.22
+```
+
+Once in the vendor directory, the script can be built into the app.
+We just need to import it into our build file.
+
+```ember-cli-build.js{+22}
+/*jshint node:true*/
+/* global require, module */
+var EmberApp = require('ember-cli/lib/broccoli/ember-app');
+
+module.exports = function(defaults) {
+  var app = new EmberApp(defaults, {
+    // Add options here
+  });
+
+  // Use `app.import` to add additional libraries to the generated
+  // output files.
+  //
+  // If you need to use different assets in different
+  // environments, specify an object as the first parameter. That
+  // object's keys should be the environment name and the values
+  // should be the asset to use in that environment.
+  //
+  // If the library that you are including contains AMD or ES6
+  // modules that you would like to import into your application
+  // please specify an object with the list of modules as keys
+  // along with the exports of each module as its value.
+  app.import('vendor/gmaps.js');
+
+  return app.toTree();
+};
+```
+
+### Accessing the Google Maps API
+
+Now that we have the maps API available to the application, we can create our map utility.
+Utility files can be generated using Ember CLI.
+
+```shell
+ember g util google-maps
+```
+
+The CLI `generate util` command will create a utility file and a unit test.
+We'll delete the unit test since we don't want to test Google code.
+Our app needs a single function, `createMapElement`, which makes use of `google.maps.Map` to create our map element, `google.maps.Geocoder` to lookup the coordinates of our location, and `google.maps.Marker` to pin our map based on the resolved location.
+
+```app/utils/google-maps.js
+import Ember from 'ember';
+
+const google = window.google;
+
+export default Ember.Object.extend({
+
+  init() {
+    this.set('geocoder', new google.maps.Geocoder());
```

Comment:
Agree that it doesn't buy me anything special, I just generally use set when setting properties for ember objects, and I think for the sake tutorial its fine to use set for setting object properties the get the reader in the mindset that in most cases they will need or want to.